rclcpp/include/rclcpp/qos.hpp

```diff
@@ -145,11 +145,39 @@ class RCLCPP_PUBLIC QoS
   // cppcheck-suppress noExplicitConstructor
   QoS(size_t history_depth);  // NOLINT(runtime/explicit): conversion constructor
 
-  /// Return the rmw qos profile.
+  /**
+   * Return the rmw qos profile where a profile is comprised of these policies:
+   *   - history (an enum from `HistoryPolicy`)
+   *   - queue size/depth
+   *   - reliability (an enum from `ReliabilityPolicy`)
+   *   - durability (an enum from `DurabilityPolicy`)
+   *   - deadline
+   *   - lifespan
+   *   - liveliness (an enum from `LivelinessPolicy`)
+   *   - lease duration
+   *   - boolean for `avoid_ros_namespace_conventions`
+   * 
+   * To retrieve an individual policy comprising the profile,
+   * use the the getter function for that policy.
+   */
```

Comment:
how about the following instead?

This version provides a brief overview while directing the user to the getter functions for more details. This way, if the rmw_qos_profile_t structure changes, the docstring remains accurate without needing updates.

```suggestion
  /**
   * Return the rmw qos profile.
   * The profile consists of various QoS policies such as history, reliability, and durability.
   * Use the corresponding getter functions to retrieve individual policies.
   */
```

Reply:
Thank you for the suggestion, it does lower maintenance cost.